src/promptflow/promptflow/batch/_base_executor_proxy.py

```diff
@@ -115,6 +117,23 @@ async def exec_aggregation_async(
         result = self._process_http_response(response)
         return AggregationResult.deserialize(result)
 
+    async def ensure_executor_startup(self, error_file):
+        """Ensure the executor service is initialized before calling the API to get the results"""
+        try:
+            await self.ensure_executor_health()
+        except ExecutorServiceUnhealthy as ex:
+            # raise the init error if there is any
+            startup_ex = self._check_startup_error_from_file(error_file) or ex
+            # TODO: will remove the destroy call after executor proxy creation is put into the run function
+            bulk_logger.error(
+                f"Failed to start up the executor due to an error: {str(startup_ex)}. "
+                "The executor will now be terminated."
+            )
+            self.destroy()
+            raise startup_ex
+        finally:
+            Path(error_file).unlink()
```

Comment:
Does the error_file always exist?

Reply:
It is created by executor proxy and its lifecycle is only in the `executor_proxy.create` function.

Reply:
Seems only C# executor proxy enforced the error_file?

Reply:
Yes. Python will raise validation exceptions directly. Maybe python will also follow this mechanism after detaching the container.